src/main/java/net/glowstone/entity/projectile/GlowThrownExpBottle.java

```diff
@@ -28,6 +31,8 @@ public void collide(LivingEntity entity) {
 
     private void spawnOrb() {
         int xp = ThreadLocalRandom.current().nextInt(9) + 3;
+        ExpBottleEvent event = EventFactory.getInstance()
+                .callEvent(new ExpBottleEvent(this, xp));
```

Comment:
Set the `xp` variable to the result of `event.getExperience()`

Reply:
But `xp` has to be calculated before calling the new event

Reply:
I mean after the event is fired. The event object is mutable by plugins, which can use setExperience to change the behaviour. 

So, after event is called, use `xp = event.getExperience();` and keep the rest as is.